test/sequential/test-repl-persistent-history.js

```diff
@@ -185,16 +228,38 @@ function runTest() {
     useColors: false,
     terminal: true
   }, function(err, repl) {
-    if (err) throw err;
+    if (err) {
+      console.error(`Failed test # ${numtests - tests.length}`);
+      throw err;
+    }
 
-    if (after) repl.on('close', after);
+    repl.once('close', () => {
+      if (repl._flushing) {
+        repl.once('flushHistory', onClose);
+        return;
+      }
 
-    repl.on('close', function() {
-      // Ensure everything that we expected was output
-      assert.strictEqual(expected.length, 0);
-      setImmediate(runTest);
+      onClose();
     });
```

Comment:
@silverwind @evanlucas can I get a last LGTM on this? this works without any even minor breaking changes now.